webui/react/src/components/kit/Facepile.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useMemo, useState } from 'react';
+
+import SelectFilter from 'components/SelectFilter';
+import { useUsers } from 'stores/users';
+import { DetailedUser } from 'types';
+import { Loadable } from 'utils/loadable';
+
+import Button from './Button';
+import css from './Facepile.module.scss';
+import UserAvatar from './UserAvatar';
+
+export interface Props {
+  editable?: boolean;
+  users?: DetailedUser[];
+  // TODO: add vertical orientation
+}
+
+const Facepile: React.FC<Props> = ({ editable = false, users = [] }) => {
+  const [showDropdown, setShowDropdown] = useState(false);
+  const [showAllAvatars, setShowAllAvatars] = useState(false);
+  const [avatars, setAvatars] = useState(users);
+  const loadableUsers = useUsers();
```

Comment:
UI Kit components are not allowed to rely on stores, this information should be passed into the component.